projects/bms_carrier/src/killswitch.c

```diff
@@ -1,34 +1,35 @@
 #include "killswitch.h"
+
+#include "bms.h"
+#include "exported_enums.h"
 #include "gpio_it.h"
-#include "log.h"
 
 static void prv_killswitch_handler(const GpioAddress *address, void *context) {
-  BatteryHeartbeatStorage *storage = context;
   GpioState state = NUM_GPIO_STATES;
   gpio_get_state(address, &state);
-  if (state == GPIO_STATE_LOW) {
-    // Falling edge - killswitch was hit
-    battery_heartbeat_raise_fault(storage, EE_BATTERY_HEARTBEAT_FAULT_SOURCE_KILLSWITCH);
-  } else {
-    // Rising edge - killswitch was released
-    battery_heartbeat_clear_fault(storage, EE_BATTERY_HEARTBEAT_FAULT_SOURCE_KILLSWITCH);
-  }
+  bool clear = state == GPIO_STATE_LOW ? true : false;
```

Comment:
Is this pattern? Because simply `clear = (state == GPIO_STATE_LOW);` should be good. I would also maybe suggest putting it directly into the below line, but I like this verbosity.

Reply:
Updated to `bool clear = (state == GPIO_STATE_LOW)`